AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLiteStorageEngineAdapterTests.swift

```diff
@@ -375,4 +375,108 @@ class SQLiteStorageEngineAdapterTests: BaseDataStoreTests {
+
+        let newVersion = "previousVersion"
+        let mockFileManager = MockFileManager()
+        mockFileManager.fileExists = true
+        mockFileManager.removeItem = { res in
+            XCTFail("Should not have called removeItem")
+        }
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testClearIfNewVersionWithMissingFile() {
+        guard let userDefaults = UserDefaults.init(suiteName: "VersionRemainsSameButFileDoesNotExist") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "previousVersion"
+        let mockFileManager = MockFileManager()
+        mockFileManager.fileExists = true
+        mockFileManager.removeItem = { res in
+            XCTFail("Should not have called removeItem")
+        }
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testClearIfNewVersionFailure() {
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionHasChanged") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+        
+        let newVersion = "newVersion"
+        let mockFileManager = MockFileManager()
+        mockFileManager.hasError = true
+        mockFileManager.fileExists = true
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTAssertEqual(error as? MockFileManagerError, MockFileManagerError.removeItemError)
```

Comment:
assert is DataStoreError.invalidDatabase

Reply:
Done